store/store.go

```diff
@@ -700,9 +700,9 @@ func (s *SnapUbuntuStoreRepository) Download(remoteSnap *snap.Info, pbar progres
 	if err != nil {
 		return "", err
 	}
-	s.setUbuntuStoreHeaders(req, "", remoteSnap.NeedsDevMode(), auther)
+	s.setUbuntuStoreHeaders(req, "", false, auther)
 
-	if err := download(remoteSnap.Name(), w, req, pbar); err != nil {
+	if err := download(name, w, req, pbar); err != nil {
 		return "", err
```

Comment:
We are doing this cleanup on failure actually in the defer func of Download() (unless I miss something).